test/tools/test_save_notes.py

```diff
@@ -0,0 +1,73 @@
+import os
+import sys
+import time
+from unittest.mock import MagicMock
+
+import pytest
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(f"{CURRENT_DIR}/../../tools")
+sys.path.append(f"{CURRENT_DIR}/../../lib")
+
+# pylint: disable=import-error,wrong-import-position
+from save_notes import save_notes
+from db import DB
+
+
+def test_invalid_string_timestamp():
+    with pytest.raises(ValueError):
+        save_notes(
+            "test_project_id",
+            [
+                {
+                    "note": "This is my note",
+                    "timestamp": 'string_instead_of_time',
+                    "detail_name": "Test_Container",
+                }
+            ],
+        )
```

Comment:
I would call an explicit assert here based on the expected error. 

an example of how I do this can be found in the test_usage_scenario.py, in test_env_variable_no_skip_or_allow()

something like:

with pytest.raises(ValueError) as err:
        [... does the thing ...]
expected_exception = ''Note timestamp did not match expected format"
assert expected_exception in str(err.value), \
        utils.assertion_info(f"Exception: {expected_exception}", str(err.value))